system/include/emscripten/wasmfs.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <sys/stat.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct Backend* backend_t;
+
+// Obtains the backend_t of a specified path.
+backend_t wasmfs_backend_by_path(char* path);
+
+// Obtains the backend_t of a specified fd.
+backend_t wasmfs_backend_by_fd(int fd);
+
+// Creates a JSFile Backend in the new file system.
+backend_t create_js_file_backend();
+
+// Can open a file using open flags, but the backend parameter is ignored.
```

Comment:
Do you mean that this function can open a file, but it ignores the backend parameter? Or that normal `open()` has that behavior?

If the former, why have the parameter if it is ignored?

If the latter, I think the comment could be improved to make it clear what it speaks about.

Reply:
I think we can resolve this by just forcing `O_CREAT` to be appended.